test/Microsoft.AspNetCore.Sockets.Common.Tests/Internal/Formatters/ServerSentEventsParserTests.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Sockets.Internal.Formatters;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Sockets.Common.Tests.Internal.Formatters
+{
+    public class ServerSentEventsParserTests
+    {
+        [Theory]
+        [InlineData("data: T\r\n\r\n", "")]
+        [InlineData("data: T\r\ndata: \r\r\n\r\n", "\r")]
+        [InlineData("data: T\r\ndata: A\rB\r\n\r\n", "A\rB")]
+        [InlineData("data: T\r\ndata: Hello, World\r\n\r\n", "Hello, World")]
+        [InlineData("data: E\r\ndata: Hello, World\r\n\r\n", "Hello, World")]
+        [InlineData("data: T\r\ndata: Hello, World\r\n\r\ndata: ", "Hello, World")]
+        public void ParseSSEMessageSuccessCases(string encodedMessage, string expectedMessage)
```

Comment:
we talked about adding some cases for standalone `\r` (i.e. `\r` not followed by `\n`)